apps/framework-cli/src/infrastructure/olap/clickhouse/diff_strategy.rs

```diff
@@ -0,0 +1,323 @@
+//! ClickHouse-specific table diffing strategy
+//!
+//! This module implements the TableDiffStrategy for ClickHouse, handling the database's
+//! specific limitations around schema changes. ClickHouse has restrictions on certain
+//! ALTER TABLE operations, particularly around ORDER BY and primary key changes.
+
+use crate::framework::core::infrastructure::table::Table;
+use crate::framework::core::infrastructure_map::{
+    ColumnChange, OlapChange, OrderByChange, TableChange, TableDiffStrategy,
+};
+
+/// ClickHouse-specific table diff strategy
+///
+/// ClickHouse has several limitations that require drop+create operations instead of ALTER:
+/// - Cannot change ORDER BY clause via ALTER TABLE
+/// - Cannot change primary key structure via ALTER TABLE
+/// - Some column type changes are not supported
+///
+/// This strategy identifies these cases and converts table updates into drop+create operations
+/// so that users see the actual operations that will be performed.
+pub struct ClickHouseTableDiffStrategy;
+
+impl TableDiffStrategy for ClickHouseTableDiffStrategy {
+    /// This function is only called when there are actual changes to the table
+    /// (column changes, ORDER BY changes, or deduplication changes).
+    /// It determines whether those changes can be handled via ALTER TABLE
+    /// or require a drop+create operation.
+    fn diff_table_update(
+        &self,
+        before: &Table,
+        after: &Table,
+        column_changes: Vec<ColumnChange>,
+        order_by_change: OrderByChange,
+    ) -> Vec<OlapChange> {
+        // Check if ORDER BY has changed
+        let order_by_changed = order_by_change.before != order_by_change.after;
```

Comment:
I wonder what happens here if they accidentally add whitespaces when the fields are actually the same. I think I would prefer the plan to show that it changed, then I fix it myself instead of moose trimming whitespaces, so the code that's checked in is exact.

Reply:
Hum - yeah I guess that's a matter of preference. 
I can leave it as is for now and we see if we get feedback on that